### PR TITLE
 Duplicate Arrays and Dictionaries when instancing scene in editor 

### LINKED ISSUE #13971: Regression: It's impossible to modify exported NodePath array content for instanced subscene
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
af27414b1c10173584539186e396668a899e06b2 Ubuntu 17.04

**Issue description:**
This is an issue that reveals quite often since I'm on 3.0 I'm not always able to reproduce it from scratch, but when it happens it's permanent. Also for the issue to happen, user need to close every scene in which given tool subscene is instanced (even if not directly).
I'm quite sure this is 3.0 related issue since I'm using array nodepaths this way since a long time, also this issue is for sure not very fresh, I tried to track it down couple of times already.
***Short video description:*** https://youtu.be/1fjAQXu49TY
(the subscene don't need to be in tool mode for this to happen as it turned out)

**Steps to reproduce:**
You can check video for quick 40s reproduction process with provided sample
1. Download reproduction project.
2. Open "ExportedNodePathArraySave" tscn
3. Add new nodepath item to 'onTest' node and fill it
4. Save the scene
5. Close it
6. Open "ExportedNodePathArraySave" tscn again and see the node path was not saved

**Link to minimal example project:**
[ExportedNodePathArraySave.zip](https://github.com/godotengine/godot/files/1528086/ExportedNodePathArraySave.zip)

### PR DESCRIPTION
**Pull request by bojidar-bg**
Fixes #13971.

### CODE REVIEW

**Comment by bojidar-bg** [comment]
Ping @hpvb about the second commit of this PR, since the original code comes from 6565cf336051c5c0dd6bc48a18d6e7024eaf6ffa.

**Comment by akien-mga** [comment]
The second commit would probably be better in its own PR to be merged quickly while the other change needs stricter review?

Also I can already see users asking for Array and Dictionary deep copy to be exposed to scripting langs. :)

**Comment by bojidar-bg** [comment]
Wait, just realized I forgot to expose the deep parameter :laughing:

File: core/array.cpp [hunk 1]
```diff
@@ -35,8 +35,8 @@
 #include "variant.h"
 #include "vector.h"
 
-struct ArrayPrivate {
-
+class ArrayPrivate {
```
Comment (hpvb) [inline core/array.cpp]: Why? In C++ a Class with only public members is perfectly equivalent to a struct in every way.

Comment (bojidar-bg) [reply core/array.cpp]: Clang complains otherwise.

**Pull request merged**